diff --git a/graphlearn_torch/csrc/cpu/random_sampler.cc b/graphlearn_torch/csrc/cpu/random_sampler.cc
--- a/graphlearn_torch/csrc/cpu/random_sampler.cc
+++ b/graphlearn_torch/csrc/cpu/random_sampler.cc
@@ -141,8 +141,8 @@ void CPURandomSampler::UniformSample(const int64_t* col_begin,
   // with replacement
   const auto cap = col_end - col_begin;
   if (req_num < cap) {
-    uint32_t seed = RandomSeedManager::getInstance().getSeed();
-    thread_local static std::mt19937 engine(seed);
+    thread_local static std::mt19937 engine(
+        RandomSeedManager::getInstance().getSeed());
     std::uniform_int_distribution<> dist(0, cap-1);
     for (int32_t i = 0; i < req_num; ++i) {
       out_nbrs[i] = col_begin[dist(engine)];
@@ -162,8 +162,8 @@ void CPURandomSampler::UniformSample(const int64_t* col_begin,
   // with replacement
   const auto cap = col_end - col_begin;
   if (req_num < cap) {
-    uint32_t seed = RandomSeedManager::getInstance().getSeed();
-    thread_local static std::mt19937 engine(seed);
+    thread_local static std::mt19937 engine(
+        RandomSeedManager::getInstance().getSeed());
     std::uniform_int_distribution<> dist(0, cap-1);
     for (int32_t i = 0; i < req_num; ++i) {
       auto idx = dist(engine);
